src/h3lib/lib/edge.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2022 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file  edge.c
+ * @brief Edge functions for manipulating (nondirected) edge indexes.
+ */
+
+#include <inttypes.h>
+#include <stdbool.h>
+
+#include "algos.h"
+#include "constants.h"
+#include "coordijk.h"
+#include "h3Index.h"
+#include "latLng.h"
+#include "vertex.h"
+
+/**
+ * Wrap the error code from a directed edge function and present
+ * undirected edge errors instead.
+ */
+H3Error wrapDirectedEdgeError(H3Error err) {
+    if (err == E_DIR_EDGE_INVALID) {
+        return E_UNDIR_EDGE_INVALID;
+    }
+    return err;
+}
+
+/**
+ * Encode the edge between the two cells in non-normalized form.
+ * @param cell1 Origin cell
+ * @param cell2 Neighboring cell
+ * @param out Non-directed edge between the two cells, non-normalized
+ */
+H3Error cellsToEdgeNonNormalized(H3Index cell1, H3Index cell2, H3Index *out) {
```

Comment:
Nit: I'm a little confused to as to what "non-normalized" means here - it just means that this function assumes normalized input? Would just `cellsToEdgeInternal` or similar make more sense?